File: backend/department-service/src/main/java/com/eems/department_service/service/DepartmentService.java
package com.eems.department_service.service;

import java.util.List;

import org.springframework.data.domain.Pageable;

import com.eems.department_service.dto.DepartmentRequestDTO;
import com.eems.department_service.dto.DepartmentResponseDTO;
import com.eems.department_service.pagination.PagedResponse;

public interface DepartmentService {

    DepartmentResponseDTO createDepartment(
            DepartmentRequestDTO requestDTO);

    DepartmentResponseDTO updateDepartment(
            Long id,
            DepartmentRequestDTO requestDTO);

    DepartmentResponseDTO getDepartmentById(Long id);

    List<DepartmentResponseDTO> getAllDepartments();

    PagedResponse<DepartmentResponseDTO> getDepartments(
            Pageable pageable,
            String search);

    void deleteDepartment(Long id);

}

File: frontend/src/services/departmentService.js
import api from "../api/axios";

const DepartmentService = {

    getAllDepartments() {
        return api.get("/departments");
    },

    getDepartmentById(id) {
        return api.get(`/departments/${id}`);
    },

    createDepartment(department) {
        return api.post("/departments", department);
    },

    updateDepartment(id, department) {
        return api.put(`/departments/${id}`, department);
    },

    deleteDepartment(id) {
        return api.delete(`/departments/${id}`);
    },

    getDepartments(
        page = 0,
        size = 10,
        sortBy = "id",
        sortDir = "asc",
        search = ""
    ) {

        return api.get("/departments/page", {
            params: {
                page,
                size,
                sortBy,
                sortDir,
                search
            }
        });

    }

};

export default DepartmentService;